libraries/AP_LocationDB/AP_LocationDB.h

```diff
@@ -0,0 +1,133 @@
+    float _radius;
+    uint8_t _flags;
+
+    uint8_t _populated_fields = 0;
+
+    bool is_field_populated(DataField option) const {
+        return (_populated_fields & (uint8_t)option) != 0;
+    }
+};
+
+class AP_LocationDB {
+public:
+    friend class AP_Filesystem_LocationDB;
+    friend class AP_OABendyRuler;
+
+    AP_LocationDB();
+
+    CLASS_NO_COPY(AP_LocationDB);
+
+    static AP_LocationDB *get_singleton() {
+        return _singleton;
+    }
+
+    void init();
+    void update();
+
+    enum class KeyDomain : uint8_t {
+        // 0 should not be used
+        MAVLINK = 1U,
+        ADSB = 2U,
+        SCRIPTING = 3U,
+    };
+
+    bool add_item(AP_LocationDB_Item item); // add item to the location database
+    bool get_item(const uint32_t key, AP_LocationDB_Item &ret); // get item with given key from location database
+    bool update_item(const uint32_t key, AP_LocationDB_Item &new_item); // update item with given key in location database
+    bool remove_item(const uint32_t key); // remove item with given key from location database
+    bool healthy() const { return _items != nullptr; } // returns true when memory is allocated to store database items
+    bool is_full() const { return _capacity == _size; }; // returns true when database is full
+    void clear(); // clear location database, i.e., remove all items
+    uint16_t size() const { return _size; } // return number of items present in location database
+    uint16_t capacity() const { return _capacity; } // return the capacity of location database
+    bool item_exists(uint32_t key); // return true when item with given key exists in the location database
+    
+    static bool is_valid_key(uint32_t key); // return true if a given key is a valid location database key
+    static uint32_t construct_key_mavlink(uint8_t sysid, uint8_t compid, uint8_t msgid); // helper method to construct location database key for a mavlink item
+    static uint32_t construct_key_adsb(uint32_t icao); // helper method to construct location database key for a adsb item
+    static uint32_t construct_key_scripting(uint32_t subkey); // helper method to construct location database key for a scripting item
+    static uint8_t short_mav_msg_id(uint32_t msgid); // shortened 8 bit message id for a mavlink message
+
+    // parameter var table
+    static const struct AP_Param::GroupInfo var_info[];
+
+    // parameters
+    AP_Int16        _database_capacity_param;                   // db capacity
+    AP_Int8         _default_flags;                             // default flags on new database item
+    AP_Int16        _item_timeout;                              // timeout for db item
+    AP_Int16        _inc_radius;                                // inclusion radius
+
+private:
+    // assert_storage_size<AP_LocationDB_Item, AP_LOCATIONDB_ITEM_SIZE> AP_LocationDB_Item UNUSED_PRIVATE_MEMBER;
+
+    uint16_t _capacity = 0;
+    uint16_t _size = 0;
+    AP_LocationDB_Item* _items = nullptr;
```

Comment:
```suggestion
```

Unless you're putting an `AP_LocationDB` on the stack (or something truly bizarre) these are implicitly zeroed.

Zeroing here just wastes flash.

Reply:
Fixed.